app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true, length: {minimum: 6}
```

Comment:
1. code `{self.email = email.downcase}` chị sửa lại như bên dưới được ko em hì
`{email.downcase!}`
Sau đó phân biệt giúp chị `downcase` và downcase!` khác nhau thế nào nhé

2. Thứ tự macro em bị ko đúng coding convention, em check lại hì
https://github.com/framgia/coding-standards/blob/master/vn/rails/standard.md#activerecord

3. Các giá trị như 50, 255, 6 em khai báo vào file settings nhé
để làm dc, em tìm hiểu `gem config in rails ` nhé